tmt/steps/report/reportportal.py

```diff
@@ -41,13 +43,56 @@ def _str_env_to_default(option: str, default: Optional[str]) -> Optional[str]:
     return str(os.getenv(env_var))
 
 
-def _filter_invalid_chars(data: str) -> str:
+class LogFilterSettings:
+    size: int
+    is_traceback: bool
+
+    def __init__(self,
+                 size: int = DEFAULT_LOG_SIZE_LIMIT,
+                 is_traceback: bool = False):
+        self.size = size
+        self.is_traceback = is_traceback
+
+
+def _filter_invalid_chars(data: str,
+                          settings: LogFilterSettings) -> str:
     return re.sub(
         '[^\u0020-\uD7FF\u0009\u000A\u000D\uE000-\uFFFD\U00010000-\U0010FFFF]+',
         '',
         data)
 
 
+def _filter_log_per_size(data: str,
+                         settings: LogFilterSettings) -> str:
+    size = len(data)
+    if settings.is_traceback:
```

Comment:
It seems to me the following lines are needed only when `if size > settings.size`, we should move them under that condition then, and construct these variables only when they are really needed.

Reply:
Done